core/src/css/typography.scss

```diff
@@ -32,6 +32,16 @@ html {
   font-family: var(--ion-font-family);
 }
 
+/**
+ * Dynamic Type is an iOS-only feature, so
+ * this should only be enabled on iOS devices.
+ */
+@supports (-webkit-touch-callout: none) {
+  html {
+    font: var(--ion-dynamic-type);
+  }
+}
```

Comment:
macOS now has Dynamic Type support. However, the base font size on macOS is 13px. While this aligns with the rest of the macOS design system, it does not align with Ionic components which are designed for mobile. As a result, enabling this on macOS would cause fonts to appear too small on desktop. As a result, the dynamic type font is only applied on iOS. `-webkit-touch-callout` detects WebKit + iOS support.